### DISCUSSION TITLE
TS Wednesday 13th December 2023 - Agenda

### DISCUSSION THREAD

**Discussion by richardbirkin** [SuperHyperInstantFutureTime/TrackShift#139] [Meetings]
Things to talk about on 13th December when @g105b is next on TS....

- [ ] 1. do we need to speed up the importer rigfht now just for Distrokid?
- [ ] 2. UPC lookup on CD Baby + Bandcamp - is it working the same? 
- [ ] 3. .xlsx file import not working Cargo Physical
- [ ] 4. merging `more-upload-types` with `master` so that alpha users can use
- [ ] 5. time estimate on adding paywall to login, including any Authwave changes needed

**Comment by g105b** [comment]
1. Speed up distrokid? No need just yet, but we should have a loading indicator message to keep people happy, with the intention of introducing the fancy record spinner soon.
2. CD Baby - no problem, because the album title has its own column (side note: it was only missing because Biff's music was taken off CD baby and moved onto Believe). Bandcamp is working OK, question was regarding UPC usages.
3. Error message needs adding when file fails to upload. I will investigate what the problem is with specific excel file (Gr. Q1 and Q2).
4. When 1, 2 and 3 are done I'll do that.
5. I've decided that all payment code should be done in TrackShift rather than a third party library, because Stripe is so nice and easy to use. Let's discuss this one in its own issue https://github.com/SuperHyperInstantFutureTime/TrackShift/issues/137.

**Comment by g105b** [comment]
I will take on CD Baby and Cargo Physical bugs first to help @richardbirkin in testing.

**Comment by g105b** [comment]
I've just been thinking about the record spinner, and how I don't like spinners, and I think I have an idea to make a perfect bit of user experience.

How about we have the good old record spinner, but above it we show the currently detected earnings, and this number keeps on incrementing until we are done processing. That way, we avoid the difficult problem of having to know what percentage through the processing we are or what time remaining there is, and we change a "spinner" from something that people want to go away to something that people want to see for as long as possible (the longer they see it, the more money they have!)
